SpatialGDK/Source/SpatialGDK/Public/Schema/UnrealObjectRef.cpp

```diff
@@ -122,8 +124,19 @@ FUnrealObjectRef FUnrealObjectRef::FromObjectPtr(UObject* ObjectValue, USpatialP
 					}
 				}
 
-				// Unresolved object. 
-				UE_LOG(LogUnrealObjectRef, Verbose, TEXT("FUnrealObjectRef::FromObjectPtr: ObjectValue is unresolved! %s"), *ObjectValue->GetName());
+				// Check if the object is a newly referenced dynamic subobject, in which case we can create the object ref if we have the entity id of the parent actor.
+				if (ObjectValue->IsA<UActorComponent>())
+				{
+					PackageMap->TryResolveNewDynamicSubobjectAndGetClassInfo(ObjectValue);
+					ObjectRef = PackageMap->GetUnrealObjectRefFromObject(ObjectValue); // This should now be valid, as we resolve the object in the line before
+					if (ObjectRef.IsValid())
+					{
+						return ObjectRef;
+					}
+				}
+
+				// Unresolved object.
+				UE_LOG(LogUnrealObjectRef, Warning, TEXT("FUnrealObjectRef::FromObjectPtr: ObjectValue is unresolved! %s"), *ObjectValue->GetName());
```

Comment:
This got an increased log severity from Verbose -> Warning, as when this fails, the reference is never set correctly, which probably breaks some game code. Will have to monitor whether this gets logged somewhere.